src/rulesets/Base/base.jl

```diff
@@ -76,7 +76,7 @@ function rrule(::typeof(hypot), z::Complex)
 end
 
 @scalar_rule fma(x, y, z) (y, x, One())
-@scalar_rule muladd(x, y, z) (y, x, One())
+@scalar_rule muladd(x, y::CommutativeMulNumber, z) (y, x, One())
```

Comment:
That's right. You have `∂Ω = Δx y + x Δy + Δz`. So depending on how `@scalar_rule` is implemented either `x` and/or `y` must be commutative. But `@scalar_rule` implements essentially `sum(∂xyz .* Δxyz)` in forward and `sum(conj.(∂xyz) .* ΔΩ)` in reverse, so `y` must commute (in forward) with `Δx` and (in reverse) with `ΔΩ`.

It might be worthwhile to restrict all arguments to be commutative here, since I suppose there could be some complication when a number that commutes under multiplication is multiplied by one that doesn't, though I can't think of an example.